packages/angular_devkit/build_angular/test/differential-loading/differential_loading_spec.ts

```diff
@@ -0,0 +1,75 @@
+/**
```

Comment:
I think there is a bit of misunderstanding here, specs are usefully next to the feature file. These specs here are usefully used to test the architect and named `large`.

These are more of an integration test rather than a unit. Unit tests should reside next to the feature.

Also, we should not interact with the File System directly, we should use the Abstraction layer as this would reduce a couple of things.

1) Makes mocking easier.
2) No need to create a physical file for each test. 
3) Have the mocks as part of the spec file hence makes the test more readable.

```
const workspaceRoot = path.join(
  devkitRoot,
  'tests/angular_devkit/build_angular/test-differential-loading/');
const host = new TestProjectHost(normalize(workspaceRoot));

describe('differential loading', () => {
  it('detects the need for ES5 if IE 9-11 has to be supported', () => {
    const path = normalize('browerlist');
    host.writeMultipleFiles({
        path: 'last 1 chrome version'
    });
    const es5SupportNeeded = isEs5BrowserSupportNeeded(workspaceRoot, path);
    expect(es5SupportNeeded).toBe(true);
  });
});
```

The `host` itself has a lot for method, that can help you with transforming a file, overriding it's content, delete it etc..., the important one is that it will never override the file on the hard drive.

Reply:
Umm actually `TestProjectHost` seems to be deprecated, @clydin what's the new preferred way to approach this?

Reply:
Still pending.

Reply:
I've tried this. While this totally makes sense, it seems that in this very case, using the virtual filesystem does not work. The reason is that browserslist has its own machanism for finding and loading the `browswerslist` config. This one isn't using the virtual filesystem.

Reply:
This should work, as the TestProjectHost writes files to disks.

See below;
<img width="752" alt="Screenshot 2019-04-02 at 17 45 13" src="https://user-images.githubusercontent.com/17563226/55416620-6a9e4980-556f-11e9-8c79-d55b315ef2ac.png">



If you need a hand let me know.

Reply:
It seems to be a Windows-related issue: `host.root()` returns sth like `/c/my-dir` instead of `c:/my-dir`. The forward slashes are no issue here.

Reply:
Yeah, that is expected if you pass a devkit path directly to the method yourself that is using FS.

Devkit paths are always posix normalised, which means on windows `c:\\` will be `/c/` which means that any interaction with FS will fail on windows. Devkit paths/methods cannot be used with direct interaction with fs. 

The projectRoot, is normalised is passed from `wco`. See https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/browser/index.ts#L127

The safest and cleaner approach would be changing the `projectRoot` to Path and using `getSystemPath`

```
export function isDifferentialLoadingNeeded(
  projectRoot: Path,
  target: ts.ScriptTarget): boolean {

   const supportES2015 = target !== ts.ScriptTarget.ES3 && target !== ts.ScriptTarget.ES5;
  const browsersList: string[] = browserslist(
    undefined, {
      path: getSystemPath(projectRoot),
    });

   return supportES2015 && !caniuse.isSupported('es6-module', browsersList.join(', '));
}
```

Reply:
Thanks!